### PR TITLE
[ui] Add context menu popover in sidebar for asset groups

### PR DESCRIPTION
**Pull request by clairelin135**
As the title:
image

Fixes issue https://github.com/dagster-io/dagster/issues/18932

Tested locally, confirming each button in the popover functioned as intended.

### CODE REVIEW

**Comment by clairelin135** [comment]
Current dependencies on/for this PR:
* `master`
  * **PR #19577** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/19577?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/19577?utm_source=stack-comment).

File: js_modules/dagster-ui/packages/ui-core/src/asset-graph/AssetGraphExplorer.tsx [hunk 1]
```diff
@@ -497,7 +497,7 @@ const AssetGraphExplorerWithData = ({
     </ShortcutHandler>
   );
 
-  const onFilterToGroup = (group: GroupLayout) => {
+  const onFilterToGroup = (group: AssetGroup) => {
```
Comment (salazarm) [inline js_modules/dagster-ui/packages/ui-core/src/asset-graph/AssetGraphExplorer.tsx:500]: Was this change necessary? If it is then this should be `AssetGroup | GroupLayout`

**Review by salazarm — APPROVED** [review]

File: js_modules/dagster-ui/packages/ui-core/src/asset-graph/sidebar/AssetSidebarNode.tsx [hunk 2]
```diff
@@ -135,6 +139,11 @@ export const AssetSidebarNode = ({
                   onChangeExplorerPath={onChangeExplorerPath}
                 />
               </ExpandMore>
+            ) : isGroupNode ? (
+              <AssetGroupPopoverMenu
```
Comment (salazarm) [inline js_modules/dagster-ui/packages/ui-core/src/asset-graph/sidebar/AssetSidebarNode.tsx:143]: <ExpandMore>

File: js_modules/dagster-ui/packages/ui-core/src/asset-graph/AssetGraphExplorer.tsx (hunk 1 above)
Comment (clairelin135) [reply js_modules/dagster-ui/packages/ui-core/src/asset-graph/AssetGraphExplorer.tsx]: Yup it was necessary since we don't have access to the `GroupLayout` from the sidebar component
